src/main/java/uk/gov/laa/ccms/caab/mapper/ApplicationMapper.java

```diff
@@ -0,0 +1,645 @@
+  @Mapping(target = "timeRecovery", source = "landAward.timeRelatedAward")
+  @Mapping(target = "liableParties", source = "landAward.otherProprietors")
+  @Mapping(target = "equity", ignore = true)
+  @Mapping(target = "awardAmount", ignore = true)
+  @Mapping(target = "opponentId", ignore = true)
+  @Mapping(target = "opponentsToSelect", ignore = true)
+  @Mapping(target = "costOrFinancial", ignore = true)
+  @Mapping(target = "effectiveDate", ignore = true)
+  @Mapping(target = "recoveryOfAwardTimeRelated", ignore = true)
+  @Mapping(target = "timeRelatedRecoveryDetails", ignore = true)
+  @Mapping(target = "triggeringEvent", ignore = true)
+  @Mapping(target = "auditTrail", ignore = true)
+  LandAward toLandAward(Award soaAward);
+
+  /**
+   * Complete the mapping for a LandAward.
+   *
+   * @param landAward - the mapping target
+   */
+  @AfterMapping
+  default void finaliseLandAward(@MappingTarget LandAward landAward) {
+    TimeRecovery timeRecovery = landAward.getTimeRecovery();
+    landAward.setRecoveryOfAwardTimeRelated(timeRecovery != null);
+    if (timeRecovery != null) {
+      timeRecovery.setAwardType(AWARD_TYPE_LAND);
+    }
+
+    // Initialise the equity value based on Valuation Amount vs Mortgage Amount Due
+    landAward.setEquity(landAward.getValuationAmount().subtract(landAward.getMortgageAmountDue()));
+  }
+
+  @Mapping(target = "ebsId", source = "awardId")
+  @Mapping(target = "dateOfOrder", source = "otherAsset.orderDate")
+  @Mapping(target = "awardType", constant = AWARD_TYPE_OTHER_ASSET)
+  @Mapping(target = "awardCode", source = "awardType")
+  @Mapping(target = ".", source = "otherAsset")
+  @Mapping(target = "valuationAmount",
+      source = "otherAsset.valuation.amount", defaultValue = "0.00")
+  @Mapping(target = "valuationCriteria", source = "otherAsset.valuation.criteria")
+  @Mapping(target = "valuationDate", source = "otherAsset.valuation.date")
+  @Mapping(target = "timeRecovery", source = "otherAsset.timeRelatedAward")
+  @Mapping(target = "liableParties", source = "otherAsset.heldBy")
+  @Mapping(target = "awardAmount", ignore = true)
+  @Mapping(target = "auditTrail", ignore = true)
+  @Mapping(target = "costOrFinancial", ignore = true)
+  @Mapping(target = "effectiveDate", ignore = true)
+  @Mapping(target = "opponentId", ignore = true)
+  @Mapping(target = "opponentsToSelect", ignore = true)
+  @Mapping(target = "recoveryOfAwardTimeRelated", ignore = true)
+  @Mapping(target = "timeRelatedRecoveryDetails", ignore = true)
+  @Mapping(target = "triggeringEvent", ignore = true)
+  OtherAssetAward toOtherAssetAward(Award soaAward);
+
+  /**
+   * Complete the mapping for an OtherAssetAward.
+   *
+   * @param otherAssetAward - the mapping target
+   */
+  @AfterMapping
+  default void finaliseOtherAssetAward(@MappingTarget OtherAssetAward otherAssetAward) {
+    TimeRecovery timeRecovery = otherAssetAward.getTimeRecovery();
+    /* TODO: To be changed once api updated */
```

Comment:
Make sure we do this!

Reply:
recoveryOfAwardTimeRelated no corrected and mapper updated